### ISSUE TITLE
Extract to (react) package?

### ISSUE THREAD

**Issue by danthegoodman1** [metrico/clickhouse-mate#42]
Would be ideal to extract this to a react package so this could be embedded in other apps.

We could then pass in the metadata about available databases (ideally being able to omit them too), tables, and schema of those tables as JSON into the editor.

The editor, table tree, and results renderer could be separate components as well. Sending the query would also be a separate component.

My intended use case is to use https://github.com/danthegoodman1/icedb as the backing store (where metadata, tables, schema, etc.) is tracked and then use ClickHouse/CHDB as the query engine. This means that the tables and schema would not be coming from clickhouse.

**Comment by danthegoodman1** [comment]
I see it's made in ace editor, but it seems like codemirror might be more friendly to custom keywords for auto completion (that's what the Stripe Sigma web query editor uses)

**Comment by danthegoodman1** [comment]
I worry it might be faster to ground-up rebuild at that point :/

**Comment by lmangani** [comment]
perhaps we should be thinking about a shared tool between icedb/chdb/etc?

**Comment by danthegoodman1** [comment]
@lmangani yeah, I think breaking it down into components too would be best. For example I really just need the syntax highlighting and completion.

https://github.com/tunnelvisionlabs/antlr4ts and https://github.com/ClickHouse/ClickHouse/tree/master/utils/antlr combined with adding the list of supported clickhouse functions would be a stellar combo
